runtime/parser/expression.go

```diff
@@ -1144,6 +1132,108 @@ func defineNestedExpression() {
 	)
 }
 
+func defineStringExpression() {
```

Comment:
Would it help to emit a different token as the starting token of the string template, other than a `lexer.TokenString`.
I was thinking, in a template string like `"hello \(name), welcome!"`, if we can emit a new token, say `lexer.TokenStringTemplatePart` for the first `"hello` part (and all subsequent string parts), then we can maybe keep the parsing logic of the strings and string templates separate?

Reply:
We could do this and keep the logic separate. However, I think handling the parsing of both is quite natural and does not affect the complexity of string templates.

Reply:
@RZhang05 can you maybe create an issue for this as a future improvement? Then we can resolve this and merge the PR.